### ISSUE TITLE
SMT crashes on solidity-stringutils

### ISSUE THREAD

**Issue by axic** [ethereum/solidity#4564]
```
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
  * frame #0: 0x00007fff5dd5cb6e libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x00007fff5df27080 libsystem_pthread.dylib`pthread_kill + 333
    frame #2: 0x00007fff5dcb81ae libsystem_c.dylib`abort + 127
    frame #3: 0x00007fff5bbb3f8f libc++abi.dylib`abort_message + 245
    frame #4: 0x00007fff5bbb4113 libc++abi.dylib`default_terminate_handler() + 241
    frame #5: 0x00007fff5cff4eab libobjc.A.dylib`_objc_terminate() + 105
    frame #6: 0x00007fff5bbcf7c9 libc++abi.dylib`std::__terminate(void (*)()) + 8
    frame #7: 0x00007fff5bbcf26f libc++abi.dylib`__cxa_throw + 121
    frame #8: 0x0000000100496f76 solc`std::__1::map<dev::solidity::VariableDeclaration const*, dev::solidity::SSAVariable, std::__1::less<dev::solidity::VariableDeclaration const*>, std::__1::allocator<std::__1::pair<dev::solidity::VariableDeclaration const* const, dev::solidity::SSAVariable> > >::at(dev::solidity::VariableDeclaration const* const&) const + 214
    frame #9: 0x00000001004824d8 solc`dev::solidity::SMTChecker::mergeVariables(std::__1::vector<dev::solidity::VariableDeclaration const*, std::__1::allocator<dev::solidity::VariableDeclaration const*> > const&, dev::solidity::smt::Expression const&, std::__1::map<dev::solidity::VariableDeclaration const*, dev::solidity::SSAVariable, std::__1::less<dev::solidity::VariableDeclaration const*>, std::__1::allocator<std::__1::pair<dev::solidity::VariableDeclaration const* const, dev::solidity::SSAVariable> > > const&, std::__1::map<dev::solidity::VariableDeclaration const*, dev::solidity::SSAVariable, std::__1::less<dev::solidity::VariableDeclaration const*>, std::__1::allocator<std::__1::pair<dev::solidity::VariableDeclaration const* const, dev::solidity::SSAVariable> > > const&) + 1880
    frame #10: 0x00000001004804ab solc`dev::solidity::SMTChecker::visit(dev::solidity::IfStatement const&) + 1419
    frame #11: 0x00000001001f9e55 solc`dev::solidity::IfStatement::accept(dev::solidity::ASTConstVisitor&) const + 37
    frame #12: 0x00000001001f9c58 solc`void dev::solidity::ASTNode::listAccept<std::__1::shared_ptr<dev::solidity::Statement> >(std::__1::vector<std::__1::shared_ptr<dev::solidity::Statement>, std::__1::allocator<std::__1::shared_ptr<dev::solidity::Statement> > > const&, dev::solidity::ASTConstVisitor&) + 440
    frame #13: 0x00000001001f9a86 solc`dev::solidity::Block::accept(dev::solidity::ASTConstVisitor&) const + 70
    frame #14: 0x00000001001f8a5a solc`dev::solidity::FunctionDefinition::accept(dev::solidity::ASTConstVisitor&) const + 250
    frame #15: 0x00000001001f71b5 solc`void dev::solidity::ASTNode::listAccept<std::__1::shared_ptr<dev::solidity::ASTNode> >(std::__1::vector<std::__1::shared_ptr<dev::solidity::ASTNode>, std::__1::allocator<std::__1::shared_ptr<dev::solidity::ASTNode> > > const&, dev::solidity::ASTConstVisitor&) + 437
    frame #16: 0x00000001001f7619 solc`dev::solidity::ContractDefinition::accept(dev::solidity::ASTConstVisitor&) const + 89
    frame #17: 0x00000001001f71b5 solc`void dev::solidity::ASTNode::listAccept<std::__1::shared_ptr<dev::solidity::ASTNode> >(std::__1::vector<std::__1::shared_ptr<dev::solidity::ASTNode>, std::__1::allocator<std::__1::shared_ptr<dev::solidity::ASTNode> > > const&, dev::solidity::ASTConstVisitor&) + 437
    frame #18: 0x00000001001f6fe3 solc`dev::solidity::SourceUnit::accept(dev::solidity::ASTConstVisitor&) const + 67
    frame #19: 0x000000010047dc0e solc`dev::solidity::SMTChecker::analyze(dev::solidity::SourceUnit const&) + 590
    frame #20: 0x000000010057021c solc`dev::solidity::CompilerStack::analyze() + 13276
    frame #21: 0x0000000100571d66 solc`dev::solidity::CompilerStack::parseAndAnalyze() + 54
    frame #22: 0x0000000100571f97 solc`dev::solidity::CompilerStack::compile() + 71
    frame #23: 0x0000000100014db0 solc`dev::solidity::CommandLineInterface::processInput() + 11360
    frame #24: 0x0000000100092c2b solc`main + 187
    frame #25: 0x00007fff5dc0c015 libdyld.dylib`start + 1
```

**Comment by axic** [comment]
After the assertion added in #4565:
```
Internal compiler error during compilation:
/Users/alex/Projects/solidity/libsolidity/formal/SMTChecker.cpp(761): Throw in function void dev::solidity::SMTChecker::mergeVariables(const vector<const dev::solidity::VariableDeclaration *> &, const smt::Expression &, const dev::solidity::SMTChecker::VariableSequenceCounters &, const dev::solidity::SMTChecker::VariableSequenceCounters &)
Dynamic exception type: boost::exception_detail::clone_impl<dev::solidity::InternalCompilerError>
std::exception::what: 
[dev::tag_comment*] = 
```

**Comment by leonardoalt** [comment]
Missing support to compound assignments

**Comment by leonardoalt** [comment]
Fixed by #6006
